core/src/main/java/hudson/WebAppMain.java

```diff
@@ -80,16 +80,19 @@
  */
 public class WebAppMain implements ServletContextListener {
 
-    // use RingBufferLogHandler class name to configure for backward compatibility
-    private static final int DEFAULT_RING_BUFFER_SIZE = SystemProperties.getInteger(RingBufferLogHandler.class.getName() + ".defaultSize", 256);
-
-    private final RingBufferLogHandler handler = new RingBufferLogHandler(DEFAULT_RING_BUFFER_SIZE) {
+    private final RingBufferLogHandler handler = new RingBufferLogHandler() {
```

Comment:
The no-arg constructor is marked as deprecated. We could use 

```suggestion
    private final RingBufferLogHandler handler = new RingBufferLogHandler(WebAppMain.getDefaultRingBufferSize()) {
```

what do you think?

Reply:
I did what was suggested. It was foolish of me to commit using a deprecated function. As were my previous mistakes.
The required functionality was achieved after all with changes in just 2 places.^^
I hope this commit does the trick.